mayastor/src/bdev/device.rs

```diff
@@ -278,18 +274,35 @@ fn io_type_to_err(
     }
 }
 
-#[inline]
-fn alloc_io_ctx(
+/// Initialize memory pool for allocating bdev I/O contexts.
+/// This must be called before the first I/O operations take place.
+pub fn bdev_io_ctx_pool_init(size: u64) {
+    BDEV_IOCTX_POOL.get_or_init(|| {
+        MemoryPool::<IoCtx>::create("bdev_io_ctx", size).expect(
+            "Failed to create memory pool [bdev_io_ctx] for bdev I/O contexts",
+        )
+    });
+}
+
+/// Allocate a bdev I/O context from the pool.
+fn alloc_bdev_io_ctx(
```

Comment:
Adding tryfrom/from and impl drop would reduce potential errors here, lets work on this on a follow up commit